static/js/settings_account.js

```diff
@@ -259,16 +259,31 @@ exports.set_up = function () {
     });
 
     $("#do_deactivate_self_button").on('click',function () {
-        $("#deactivate_self_modal").modal("hide");
-        channel.del({
-            url: '/json/users/me',
-            success: function () {
-                window.location.href = "/login";
-            },
-            error: function (xhr) {
-                ui_report.error(i18n.t("Error deactivating account"), xhr, $('#account-settings-status').expectOne());
-            },
+        $("#do_deactivate_self_button .loader").css('display', 'inline-block');
+        $("#do_deactivate_self_button span").hide();
+        $("#do_deactivate_self_button img").each(function () {
+            var $img = $(this);
+            var imgURL = $img.attr('src');
+            $.get(imgURL, function (data) {
+                var $svg = $(data).find('svg');
+                $svg.find("rect").css("fill", "#000");
+                $img.replaceWith($svg);
+            }, 'xml');
```

Comment:
What's the story with this block?  It's not clear to me why we should need to do a `$.get` call here.

Reply:
(Also note that I just lint-banned using `$.get`, since our style is to always use the `channel` module for network calls.  But I think we shouldn't be using a network call here in the first place).

Reply:
The same spinner as we used in 'Log In' was not visible inside the 'Deactivate Now' button. Therefore, the colour of the spinner needed to be changed and for that I had to make the svg image inline. That's was the `$.get` is for and the CSS is changed inside the block.

Reply:
There has to be a cleaner way to do this that doesn't involve an extra network request here; can you find one?  Feel free to post in #frontend to get help.